### PR TITLE
feat(export): added possibility of exporting merged CSV files

### PR DESCRIPTION
**Pull request by LAMM26**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines:https://github.com/infra-geo-ouverte/igo2/blob/master/.github/CONTRIBUTING.md#git-commit-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
When exporting multiple layers, if the comma separated CSV or Excel compatible CSV option is selected, one CSV file is exported per layer.

**What is the new behavior?**
- When the comma separated CSV or Excel compatible CSV option and two or more layers are selected, a toggle gives the option to combine all layers into one file.
- When the above toggle is toggled, another toggle gives the option to add an empty line between layers in the file.
- A layer column has been added to CSV files.
- When the user uses the Spatial Filter tool and clicks the Export results button, the "Zone" layer is unselected by default.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

**If this PR contains a breaking change, please describe the impact and migration path for existing applications:**

**Other information**:

### CODE REVIEW

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts [hunk 1]
```diff
@@ -459,11 +459,19 @@ export class ImportExportComponent implements OnDestroy, OnInit {
...
-      let filename = lay.title;
-      if (data.name !== undefined) {
-        filename = data.name;
+      if (!(data.format === ExportFormat.CSVsemicolon || data.format === ExportFormat.CSVcomma) || ! data.combineLayers) {
+        filename = lay.title;
+        if (data.name) {
+          filename = data.name;
+        }
+      } else {
+        filename = "combinedLayers";
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/lib/import-export/import-export/import-export.component.ts:474]: Translation would be better

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts [hunk 2]
```diff
@@ -552,36 +562,88 @@ export class ImportExportComponent implements OnDestroy, OnInit {
...
+
+                this.loading$.next(false);
             }
-        ));
+          ));
+        }
       }
-    });
+    };
+    if ((data.format === ExportFormat.CSVsemicolon || data.format === ExportFormat.CSVcomma) && data.combineLayers) {
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/lib/import-export/import-export/import-export.component.ts:625]: Should be put into else condition (line 561) ?

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts [hunk 3]
```diff
@@ -552,36 +562,88 @@ export class ImportExportComponent implements OnDestroy, OnInit {
...
+      this.exportService.export(featuresCSV, data.format, filename, data.encoding, this.map.projection)
+      .subscribe(
+        () => {},
+        (error: Error) => this.onFileExportError(error),
+        () => {
+          this.onFileExportSuccess();
+
+          featuresCSV.forEach(feature => {
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/lib/import-export/import-export/import-export.component.ts:633]: Create method instead ? Could be also use for geomType at line 609

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts (hunk 1 above)
Comment (LAMM26) [reply packages/geo/src/lib/import-export/import-export/import-export.component.ts:474]: Done

File: packages/geo/src/lib/import-export/import-export/import-export.component.ts (hunk 2 above)
Comment (LAMM26) [reply packages/geo/src/lib/import-export/import-export/import-export.component.ts:625]: Done

**Review by PhilippeLafreniere18 — APPROVED** [review]

**Pull request merged**